lib/discovery.js

```diff
@@ -456,5 +389,15 @@ function mixinDiscovery(MsSQL) {
       default:
         return 'String';
     }
-  }
+  };
+
+  MsSQL.prototype.setDefaultOptions = function(options) {
```

Comment:
Only MySQL has a full implementation of the function so for consistency, we are throwing the error in the base connector for now, but again we will do some digging to improve this implementation in version 2.0.